### ISSUE TITLE
Will kv secret engine expire after somedays if i don't specify max_lease_ttl option

### ISSUE THREAD

**Issue by adityabyreddy74** [hashicorp/vault#8715]
Hi,

What happens if I don't pass max_lease_ttl option while enabling a secret engine?

For eg. vault secrets enable -path=secret -version=2 kv

Will the secrets under `secret` path available for lifelong? Or they get expire after somedays?

**Current secret engine configuration**
`
  "secret/": {
    "uuid": "b450b90a-af42-7a65-014f-6bc01db5cc33",
    "type": "kv",
    "description": "",
    "accessor": "kv_6cff95ed",
    "config": {
      "default_lease_ttl": 0,
      "max_lease_ttl": 0,
      "force_no_cache": false
    },
    "options": {
      "version": "2"
    },
    "local": false,
    "seal_wrap": false
  }
`

**Vault server configuration**

`
{
    "listener": {
        "tcp": {
            "address": "[::]:8200",
            "cluster_address": "[::]:8201",
            "tls_cert_file": "/vault/tls/tls.crt",
            "tls_cipher_suites": "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA",
            "tls_disable": false,
            "tls_key_file": "/vault/tls/tls.key",
            "tls_prefer_server_cipher_suites": true
        }
    },
    "storage": {
        "postgresql": {
            "connection_url": "postgres://postgres:postgres@postgresql:5432/vault?sslmode=disable",
            "ha_enabled": "true",
            "ha_table": "vault_ha_locks"
        }
    },
    "ui": false
}
`

**Comment by tyrannosaurus-becks** [comment]
Hi @adityabyreddy74  ! Thanks for opening this issue. Are you aware of our forum located here: https://discuss.hashicorp.com/c/vault/30? If possible, we appreciate it when folks ask questions there because we generally use Github for bugs and enhancement requests.

Regarding this question, we actually get it fairly regularly. Although at a back-end level we do indeed offer the `max_lease_ttl` option on all secrets engines, it only has an effect when the secrets expire. In the KV V2 secrets engine, secrets never expire, so the config option has no effect. It is simply there because we standardized the options across all back-ends. 

There are additional [api docs here](https://www.vaultproject.io/api-docs/secret/kv/kv-v2/) in case they're of help, and you'll note that ttl's aren't mentioned there.

Thanks again! Closing because there doesn't seem to be anything further to do on our side.
